hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/replication/VerifyReplication.java

```diff
@@ -788,6 +852,23 @@ private static void printUsage(final String errorMsg) {
         + "2181:/cluster-b \\\n" + "     TestTable");
   }
 
+  private static ExecutorService buildReCompareExecutor(int maxThreads, Mapper.Context context) {
+    return new ThreadPoolExecutor(0, maxThreads, 1L, TimeUnit.SECONDS, new SynchronousQueue<>(),
+      buildRejectedReComparePolicy(context));
+  }
+
+  private static CallerRunsPolicy buildRejectedReComparePolicy(Mapper.Context context) {
+    return new CallerRunsPolicy() {
+      @Override
+      public void rejectedExecution(Runnable runnable, ThreadPoolExecutor e) {
+        LOG.debug("Re-comparison execution rejected. Running in main thread.");
+        context.getCounter(Counters.REJECTED_RE_COMPARES).increment(1);
```

Comment:
this counter makes it sound like the recompare didn't happen. maybe we should call it like `MAIN_THREAD_RE_COMPARES` or something?